src/renderer/Renderer.js

```diff
@@ -187,13 +188,30 @@ export default class Renderer {
                     continue;
                 }
 
-                for (let j = 0; j < viewportExpressionsLength; j++) {
-                    viewportExpressions[j].accumViewportAgg(feature);
-                }
+                // Pass the rawFeature to all viewport aggregations
+                this._accumViewportAggregations(viewportExpressions, feature, renderLayer);
             }
         });
     }
 
+    _accumViewportAggregations (viewportExpressions, rawFeature, renderLayer) {
+        const viewportExpressionsLength = viewportExpressions.length;
+
+        for (let j = 0; j < viewportExpressionsLength; j++) {
+            const currentViewportExp = viewportExpressions[j];
+            if (currentViewportExp instanceof ViewportFeatures) {
+                const interactivityFeature = new Feature(rawFeature,
```

Comment:
I think the place for this is in `viewportFeatures`, although I guess that some `renderLayer` parameter will need to be passed to the expression (maybe in the reset function?).

On top of that, I think this code will perform poorly since `viewportFeatures` may return many features and the `Feature` constructor is not lightweight. I hope that the same trick that is being used in `viewportFeatures` to mitigate memory allocation overhead can be used for this too.

Reply:
@davidmanzanares That was my first intention, as it feels more natural. 

I was using `accumViewportAgg (interactivityFeature, renderLayer)` and it looked well, but bringing `Feature` into `viewportFeatures` breaks the current build (and tests). It seems to be causing a circular dependency problem, regarding to `expressions`. You can reproduce it just by adding this to viewportFeatures code:
`import Feature from '../../../interactivity/feature';`
We can try to dig more into that or just keep it in `Renderer` for now.

On the other hand, I'll try to mitigate memory allocation by transforming object construction.